src/as2fm/scxml_converter/scxml_entries/scxml_executable_entries.py

```diff
@@ -542,3 +562,35 @@ def as_plain_execution_body(
     if exec_body is None:
         return None
     return [entry.as_plain_scxml(ros_declarations) for entry in exec_body if not is_comment(entry)]
+
+
+def add_targets_to_scxml_send(
+    exec_body: Optional[ScxmlExecutionBody], events_to_automata: EventsToAutomata
+) -> Optional[ScxmlExecutionBody]:
+    """For each ScxmlSend in the body, generate instances containing the target automata."""
+    if exec_body is None:
+        return None
+    new_body: ScxmlExecutionBody = []
+    for entry in exec_body:
+        if isinstance(entry, ScxmlIf):
+            if_conditionals = []
+            for cond, cond_body in entry.get_conditional_executions():
+                if_conditionals.append(
+                    (cond, add_targets_to_scxml_send(cond_body, events_to_automata))
+                )
+            else_body = add_targets_to_scxml_send(entry.get_else_execution(), events_to_automata)
+            new_body.append(ScxmlIf(if_conditionals, else_body))
+        elif isinstance(entry, ScxmlSend):
+            target_automata = events_to_automata.get(entry.get_event(), {"NONE"})
```

Comment:
@EnricoGhiorzi on this one I am not entirely sure: there are cases in which we send an event, but we define no transition for it.
This might be desired in case the developer wants to expose a specific information from a single automaton only for referring to it in a property.
For now, I decided to add a "NONE" target for such cases, but we can find other alternatives

Reply:
I see the use for such a feature, but I worry about aligning the model with the implementation. Would this be how the real system implements such a feature? In other words, is it possible to monitor a topic that has a publisher but no subscribers? (I guess the monitor itself is a subscriber, but that is not modelled.)

Reguarding the "NONE" syntax, I looked up the SCXML specification but is provided no insight: there is no way to send an event into the void, and having a non-existing target would raise an error.

Another option could be to automatically create such a "NONE" automaton and just let it process any event it receives.

Reply:
To monitor a topic, you need to subscribe to it, and as you mentioned above, the monitor will do exactly that.
If you are fine with it, you can assume to have the NONE automaton that acts as a receiver for all events (though I am not sure it makes sense to auto-generate it: maybe it is easier to have a script removing the send events with a NONE target)

Reply:
> maybe it is easier to have a script removing the send events with a NONE target

I don't get it: if we remove these sends, the events will not be observable in SCAN either. I think we either need to send the events into the void (as it is now) and allow that in SCAN, or to generate a target NONE automaton, either at AS2FM or SCAN level, possibly consistently with the actual implementation. @SofiaFaraci might have some insight on this.

Reply:
My comment about (optionally) removing the NONE events was only for executing system outside of SCAN. Otherwise, I would be happy to keep it as is and allow the NONE target in SCAN, if that works for you.

Reply:
>I think we either need to send the events into the void (as it is now) and allow that in SCAN, or to generate a target NONE automaton, either at AS2FM or SCAN level, possibly consistently with the actual implementation. @SofiaFaraci might have some insight on this.

In the actual implementation there is no need for a target for the send tag, the events are directly handled by the generated C++ code of the SMs that sends the event.